scripts/conflicts/diff.js

```diff
@@ -20,37 +26,90 @@ if (!server || argv.h || argv.help) {
 
 console.log(`Generating diffs for conflicts on ${server}`);
 
+
 const DB = new PouchDB(server);
 
-DB.query('medic-conflicts/conflicts')
+
+const writeToFile = function(path, content) {
+  const writeStream = fs.createWriteStream(path);
+  writeStream.write(JSON.stringify(content, null, 3));
+  writeStream.close();
+};
+
+DB.query('medic-conflicts/conflicts', {reduce:false})
   .then(conflicts => {
     console.log(`Found ${conflicts.rows.length} conflicts`);
 
+    const mainConflictFilesDirectory = './doc-conflicts';
     let ps = Promise.resolve();
+    if (!fs.existsSync(mainConflictFilesDirectory)){
+      try {
+        fs.mkdirSync(mainConflictFilesDirectory);
+      }catch(err){
+        console.log(`Error while creating main conflict folder: ${err}`);
+      }
+    }
```

Comment:
Please pull all of this in a separate function. 
Also, consider the scenario where the folder already exists and you have previous conflicts saved there. If some of those were already resolved and the user didn't clean them up, it might end up confusing. I would fail the script if the folder was not empty, just to make sure that the resulting files are always consistent and correct.

Reply:
Thank you for the suggestions. The program now exits if the directory is not empty. Also extracted the code to create directory name and directory to another separate function.